src/pmi/src/pmi_v1.c

```diff
@@ -324,7 +340,11 @@ PMI_API_PUBLIC int PMI_Barrier_group(const int *group, int count, const char *ta
     }
 
     char *group_str;
-    if (group == PMI_GROUP_WORLD) {
+    if (group == PMI_GROUP_WORLD && !tag && !PMIU_is_threaded) {
```

Comment:
isn't `PMIU_is_threaded=1` unconditional currently? should we disable after getting the version back from the server in `PMI_Init`?

Reply:
Good suggestion!

Reply:
I suppose that disabling thread support based on server version could allow sessions apps using the world process group to function with gforker?

Reply:
Please review the last commit.

Reply:
> I suppose that disabling thread support based on server version could allow sessions apps using the world process group to function with gforker?

```
if (group == PMI_GROUP_WORLD && !tag && !PMIU_is_threaded) {
    /* Backward-compatible PMI_Barrier */
    pmi_errno = PMI_Barrier();
}
```
`MPI_Comm_create_from_group` will have `tag` set, so it still won't work    But setting `MPIR_CVAR_PMI_DISABLE_GROUP` (https://github.com/pmodels/mpich/pull/7354/commits/dd1a84e139ece2f21b19ee02c9da9da7b9c17433)  will fallback to `PMI_Barrier` and make sessions over world work